libs/natparse/src/main/java/org/amshove/natparse/lexing/Lexer.java

```diff
@@ -891,20 +910,58 @@ private boolean mightBeDataFormat(String possibleDataFormat)
 		return floatingPointCount < 2;
 	}
 
+	private boolean consumeNaturalHeader()
+	{
+		if (isAtLineStart() && (isSingleAsteriskComment() || isInlineComment()))
+		{
+			//NOOP
+		}
+		else
+		{
+			return false;
+		}
+
+		scanner.start();
+		while (!isLineEnd() && !scanner.isAtEnd())
+		{
+			scanner.advance();
+		}
+		String s = scanner.lexemeText().stripTrailing();
+
+		if (inSourceHeader)
+		{
+			if (s.contains("* <Natural Source Header"))
+			{
+				sourceHeader = new NaturalHeader(mode, lineIncrement);
+				sourceHeaderDone = true;
+			}
+			else
```

Comment:
The `else` here and after checking for `* :Mode` can be removed, so that all three ifs are on the same level.
This reduces indentation and complexity.

Reply:
... but slightly increases performance. ;)

Reply:
Haha, and now having less code, I am below 80% code coverage 👅